src/core/include/openvino/pass/pattern/op/pattern.hpp

```diff
@@ -85,7 +85,7 @@ class OPENVINO_API Pattern : public Node {
     Pattern(const OutputVector& patterns) : Pattern(patterns, nullptr) {}
 
     std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& /* new_args */) const override {
-        OPENVINO_THROW("Uncopyable");
+        OPENVINO_THROW_NORETURN("Uncopyable");
     }
```

Comment:
let's move this method to cpp to prevent inlining of this method with exceptions.